src/Microsoft.AspNetCore.ResponseCompression/CompressionProviderCollection.cs

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.ObjectModel;
+#if NETSTANDARD1_3
+using System.Reflection;
+#endif
+
+namespace Microsoft.AspNetCore.ResponseCompression
+{
+    /// <summary>
+    /// A Collection of ICompressionProvider's that also allows them to be instantiated from DI.
+    /// </summary>
+    public class CompressionProviderCollection : Collection<ICompressionProvider>
+    {
+        /// <summary>
+        /// Adds a type representing an <see cref="ICompressionProvider"/>.
+        /// </summary>
+        /// <remarks>
+        /// Provider instances will be created through dependency injection.
+        /// </remarks>
+        public void Add<T>()
```

Comment:
Gotcha. For some reason I thought you were doing `Activator.CreateInstance` in the factory.